test/e2e/network/service.go

```diff
@@ -2027,6 +2027,225 @@ var _ = common.SIGDescribe("Services", func() {
+			framework.Logf("Couldn't detect KubeProxy mode - test failure may be expected: %v", err)
+		}
+
+		serviceName := "svc-itp"
+		ns := f.Namespace.Name
+		servicePort := 80
+
+		ginkgo.By("creating a TCP service " + serviceName + " with type=ClusterIP and internalTrafficPolicy=Local in namespace " + ns)
+		jig := e2eservice.NewTestJig(cs, ns, serviceName)
+		jig.ExternalIPs = true
+
+		local := v1.ServiceInternalTrafficPolicyLocal
+		svc, err := jig.CreateTCPService(func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{
+				{Port: 80, Name: "http", Protocol: v1.ProtocolTCP, TargetPort: intstr.FromInt(80)},
+			}
+			svc.Spec.InternalTrafficPolicy = &local
+		})
+		framework.ExpectNoError(err)
+		defer func() {
+			framework.Logf("Cleaning up the webserver service")
+			err := cs.CoreV1().Services(ns).Delete(context.TODO(), serviceName, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete service: %s in namespace: %s", serviceName, ns)
+		}()
+		serviceIP := svc.Spec.ClusterIP
+		framework.Logf("internal-traffic-policy-test cluster ip: %s", serviceIP)
+		ginkgo.By("Picking 2 Nodes to test whether internal traffic policy is respected")
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 2)
+		framework.ExpectNoError(err)
+		nodeCounts := len(nodes.Items)
+		if nodeCounts < 2 {
+			e2eskipper.Skipf("The test requires at least two ready nodes on %s, but found %v", framework.TestContext.Provider, nodeCounts)
+		}
+
+		ginkgo.By("Creating 2 webserver pods to be part of the TCP service")
+		webserverPod0 := e2epod.NewAgnhostPod(ns, "echo-hostname-0", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod0.Labels = jig.Labels
+		webserverPod0.Spec.NodeName = nodes.Items[0].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod0, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod0.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		webserverPod1 := e2epod.NewAgnhostPod(ns, "echo-hostname-1", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod1.Labels = jig.Labels
+		webserverPod1.Spec.NodeName = nodes.Items[1].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod1.Name, f.Namespace.Name, framework.PodStartTimeout))
+		defer func() {
+			framework.Logf("Cleaning up the echo server pod")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod0.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod0.Name)
+			err = cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod1.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod1.Name)
+		}()
```

Comment:
I think that these are deleted by the framework, it deletes everything in the namespace ... I saw it in more of tests, maybe are reminiscences ?

Reply:
yeah that's a good point, I can remove these